### PR TITLE
add catch in OnRecoveryMessage

### PR DESCRIPTION
**Pull request by KickSeason**
Fix https://github.com/neo-project/neo/pull/1803#issuecomment-670361121

We have a risk that one validator can easily stop all c# consensus nodes with put out of range validator index in some payloads in recovery message.

* Catch`IndexOutOfRangeException`

### CODE REVIEW

File: src/neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -371,6 +371,10 @@ private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage
                         if (ReverifyAndProcessPayload(commitPayload)) validCommits++;
                 }
             }
+            catch (IndexOutOfRangeException e)
+            {
+                Log($"invalid recovery message with illegal validator index. {e}");
```
Comment (shargon) [inline src/neo/Consensus/ConsensusService.cs:376]: I think that it's better to log any `Exception`

Comment (KickSeason) [reply src/neo/Consensus/ConsensusService.cs]: done.

**Comment by erikzhang** [comment]
I prefer to make a real fix, not just catch the exceptions and ignore them.

**Comment by erikzhang** [comment]
See #1837

**Comment by KickSeason** [comment]
> See #1837

Will do.

**Pull request closed without merging**